saleor/vendor/models.py

```diff
@@ -0,0 +1,65 @@
+from django.conf import settings
+from django.db import models
+
+from saleor.core.permissions import VendorPermissions
+
+# Create your models here.
+from ..warehouse.models import Allocation, Warehouse
+
+
+class Vendor(models.Model):
+    """Model definition for Vendor."""
+
+    # TODO: Define fields here
+    user = models.OneToOneField(
+        settings.AUTH_USER_MODEL,
+        blank=True,
+        null=True,
+        related_name="vendor_user",
+        on_delete=models.CASCADE,
+    )
+    # here the ideas clash
+    # one django doesn't have onetomany only manytoone
+    # so either I make vendor foreign key in warehouse
+    # or here is a dirty little trick to make manytomany
+    # into one to many
+
+    allocation = models.ManyToManyField(Allocation, related_name="vendor_allocation")
+    shop_name = models.CharField(max_length=256)
+
```

Comment:
a slug filed would be nice as in warehouse

Reply:
@Grunze1s the slug has been added in the latest commit. please review the changes and approve them.